dattri/metrics/britteness.py

```diff
@@ -0,0 +1,115 @@
+    device: torch.device,
+    batch_size: int = 64,
+) -> Optional[int]:
+    """Calculate smallest k make a test data flip.
+
+    This function calculate the brittleness metric by determining the smallest subset of
+    training data whose removal causes the test sample's prediction to flip.
+
+    Args:
+        test_data (tuple): A tuple containing the image and label of the test sample.
+        train_loader (DataLoader): DataLoader for the training data.
+        train_data_indices (list[int]): Indices of the training data used.
+        scores (torch.Tensor):
+            Data attribution scores associated with each training sample.
+        steplength (tuple): Tuple containing start, end, and step size for the k range.
+        train_func (function): Function to retrain the model on the modified dataset.
+        device (torch.device): The device (CPU or GPU) to perform computations on.
+        batch_size (int):
+            The batch size used when creating new DataLoaders. Default is 64.
+
+    Returns:
+        int or None: The smallest number of influential training points whose removal
+        flips the test point's prediction, or None if no such k exists.
+    """
+    start, end, step = steplength
+    k_values = list(range(start, end, step))
+    if isinstance(scores, torch.Tensor):
+        scores = scores.cpu().numpy() if scores.is_cuda else scores.numpy()
+    sorted_indices = np.argsort(scores)[::-1]
+    for k in tqdm(
+        k_values,
+        desc="Calculating brittleness",
+        leave=False,
+        ):
+        highest_score_indices = sorted_indices[:k]
+        if check_if_flip(test_data,
+                         train_loader,
+                         train_data_indices,
+                         highest_score_indices,
+                         train_func,
+                         device,
+                         batch_size):
+            return k
+    return None
+
+
+def check_if_flip(
+    test_data: Tuple[torch.Tensor, torch.Tensor],
+    train_loader: DataLoader,
+    train_data_indices: List[int],
+    indices_to_remove: List[int],
+    train_func: Callable[[DataLoader], torch.nn.Module],
+    device: torch.device,
+    batch_size: int,
+) -> bool:
+    """Flip check for brittleness prediction.
+
+    This function weill check if the prediction for a test sample flips after
+    removing certain training data.
+
+    Args:
+        test_data (tuple): The test data consisting of an image and its label.
+        train_loader (DataLoader): DataLoader for the training data.
+        train_data_indices (list[int]):
+            List of indices indicating orignial data the used training data.
+        indices_to_remove (list[int]):
+            Indices of the training data to remove.
+        train_func (function): Function to train the model on the modified dataset.
+        device (torch.device): The device for computation.
+        batch_size (int): Batch size for the DataLoader of the new training set.
```

Comment:
should we add a retraining_num here to do the retraining multiple times to handle the variance?

Reply:
When we call the train_mnist or our train function in dattri, there will be a fixed seed (default as 0). In the experiment file, I tried it multiple times and the result seems not change. But for users cutomize train function, we can discuss whether we can add a parameter for retraining or use interpolate.

Reply:
You mean https://github.com/TRAIS-Lab/dattri/blob/main/dattri/benchmark/datasets/mnist/mnist_lr.py#L18-L53 the `seed` parameter here does not work? Do you mean the logits keep the same or just the binary pred==label.